### PR TITLE
BC-1312 - allow admin to start user migration

### PR DESCRIPTION
**Pull request by virgilchiriac**
# Short Description

## Links to Ticket and related Pull-Requests
- https://ticketsystem.hpi-schul-cloud.org/browse/BC-1312
- https://github.com/hpi-schul-cloud/schulcloud-client/pull/2789
- https://github.com/hpi-schul-cloud/schulcloud-server/pull/3355/
- https://github.com/hpi-schul-cloud/dof_app_deploy/pull/172

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/pages/administration/migration/index.unit.ts [hunk 1]
```diff
@@ -3,13 +3,16 @@ import { mount, shallowMount } from "@vue/test-utils";
...
 import SchoolsModule from "@store/schools";
+import EnvConfigModule from "@/store/env-config";
 
 declare var createComponentMocks: Function;
 
 const $theme = {
 	short_name: "instance name",
 };
 
+//EnvConfigModule.setEnvs({ FEATURE_USER_MIGRATION_ENABLED: true });
```
Comment (virgilchiriac) [inline src/pages/administration/migration/index.unit.ts:14]: remove comment

File: src/components/organisms/administration/importUsers.vue [hunk 2]
```diff
@@ -26,7 +26,7 @@
 			{{ $t("pages.administration.migration.cannotStart") }}
```
Comment (murat-merdoglu-hpi) [inline src/components/organisms/administration/importUsers.vue:26]: The filename of a component should start with a capital letter.

**Review by murat-merdoglu-hpi — CHANGES_REQUESTED** [review]

File: src/pages/administration/migration/index.unit.ts [hunk 3]
```diff
@@ -128,11 +131,14 @@ const schoolMock = {
 	isTeamCreationByStudentsEnabled: false,
 };
 
+
 window.scrollTo = jest.fn();
 
-describe("User Migration / Index", () => {
-	beforeEach(() => {
+describe.only("User Migration / Index", () => {
```
Comment (virgilchiriac) [inline src/pages/administration/migration/index.unit.ts:137]: remove only

File: locale/en.json [hunk 4]
```diff
@@ -338,12 +338,15 @@
...
+	"pages.administration.migration.tutorialWait": "Please note, once the school migration process starts, it can take up to 1 hour to fetch the data. After this, you will be able to continue to the next step.",
+	"pages.administration.migration.waiting": "waiting for data sync...",
```
Comment (MartinSchuhmacher) [inline locale/en.json:349]: to be consistent I would start with capital "Waiting" here

File: locale/es.json [hunk 5]
```diff
@@ -339,9 +339,12 @@
...
+	"pages.administration.migration.tutorialWait": "Tenga en cuenta que, una vez que se inicie el proceso de migración de la escuela, puede tardar hasta 1 hora en obtener los datos. Después de esto, podrá continuar con el siguiente paso.",
+	"pages.administration.migration.waiting": "esperando la sincronización de datos...",
```
Comment (MartinSchuhmacher) [inline locale/es.json:347]: same here with "Esperando"

File: src/store/schools.ts [hunk 6]
```diff
@@ -49,6 +50,8 @@ function transformSchoolClientToServer(school: any): School {
 	stateFactory: true,
 })
 export class Schools extends VuexModule {
+	private _importUserApi?: UserImportApiInterface;
```
Comment (MartinSchuhmacher) [inline src/store/schools.ts:53]: is the underscore important for any syntax convention?

**Review by MartinSchuhmacher — COMMENTED** [review]
I was thinking about more testcases, but none came into my mind which weren´t there already. So only some small remarks

File: src/store/schools.ts (hunk 6 above)
Comment (virgilchiriac) [reply src/store/schools.ts:53]: is for consistency, as the same _ name is used in more stores files

Comment (murat-merdoglu-hpi) [reply src/store/schools.ts:53]: The underscored variable name is more often used for singleton design patterns. It is not necessary but common usage is like this.

File: src/components/organisms/administration/importUsers.vue (hunk 2 above)
Comment (virgilchiriac) [reply src/components/organisms/administration/importUsers.vue:26]: done

**Review by murat-merdoglu-hpi — APPROVED** [review]

**Pull request merged**